sdk/java-sdk-spring/src/main/scala/kalix/spring/impl/RestKalixClientImpl.scala

```diff
@@ -84,16 +94,31 @@ final class RestKalixClientImpl(messageCodec: JsonMessageCodec) extends KalixCli
   }
 
   override def get[R](uriStr: String, returnType: Class[R]): DeferredCall[Any, R] = {
-    matchMethodOrThrow(HttpMethods.GET, uriStr) { httpDef =>
+    runGet(uriStr, Map.empty, new LinkedMultiValueMap(), returnType)
+  }
+
+  def runGet[R](
+      pathTemplate: String,
+      pathVariables: Map[String, String],
+      queryParams: MultiValueMap[String, String],
+      returnType: Class[R]): DeferredCall[Any, R] = {
+
+    val encodedPathVariables = pathVariables.view.mapValues(URLEncoder.encode(_, StandardCharsets.UTF_8)).toMap
```

Comment:
it's a bit twisted, but I need to encode variables here to use them for Uri constructor, but for the webclient I need to pass not encoded ones, otherwise, they will be encoded twice :) I'm planning to reorganize this in the following PRs.